tests/rptest/scale_tests/audit_log_test.py

```diff
@@ -0,0 +1,412 @@
+# Copyright 2023 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import time
+import threading
+from ducktape.mark import ok_to_fail
+from rptest.clients.types import TopicSpec
+from rptest.clients.rpk import RpkTool
+from rptest.services.admin import Admin
+from rptest.services.cluster import cluster
+from rptest.services.redpanda import SaslCredentials
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.utils.scale_parameters import ScaleParameters
+from rptest.services.kgo_repeater_service import KgoRepeaterService, repeater_traffic
+from rptest.services.redpanda import SaslCredentials, SecurityConfig, MetricsEndpoint, MetricSample
+from rptest.tests.cluster_config_test import wait_for_version_sync
+from rptest.util import wait_until_result
+from ducktape.errors import TimeoutError
+
+# How much memory to assign to redpanda per partition. Redpanda will be started
+# with MIB_PER_PARTITION * PARTITIONS_PER_SHARD * CORE_COUNT memory
+DEFAULT_MIB_PER_PARTITION = 4
+
+# How many partitions we will create per shard: this is the primary scaling
+# factor that controls how many partitions a given cluster will get.
+PARTITIONS_PER_SHARD = 1000
+
+
+class AuditLogTestSecurityConfig(SecurityConfig):
+    """
+    Hardcoded security parameters for the audit scale tests. SASL will be
+    the authentication method of choice
+    """
+    sasl_credentials = SaslCredentials('rob', '123rob456', 'SCRAM-SHA-256')
+
+    def __init__(self):
+        super(AuditLogTestSecurityConfig, self).__init__()
+        self.enable_sasl = True
+        self.kafka_enable_authorization = True
+        self.endpoint_authn_method = 'sasl'
+        self.tls_provider = None
+
+
+class AuditBenchmarkResults():
+    def __init__(self, time_elapsed: float, final_totals: tuple[int, int],
+                 latencies: tuple[float, float, float], message_size: int):
+        self.time_elapsed = time_elapsed
+        self.messages_produced = final_totals[0]
+        self.messages_consumed = final_totals[1]
+        self.message_size = message_size
+        self.p50 = latencies[0]
+        self.p90 = latencies[1]
+        self.p99 = latencies[2]
+
+    def _mbps(self, b: float) -> float:
+        return (b / self.time_elapsed) / (1024 * 1024.0)
```

Comment:
question: why `1024 * 1024.0` and not `1024.0 * 1024.0`?

Reply:
took it from elsewhere, effectively should be the same